src/create-reminder-signup/lib/fetch-with-retry.ts

```diff
@@ -0,0 +1,15 @@
+import { Response } from 'node-fetch';
+
+export async function fetchWithRetry(
+	fetcher: () => Promise<Response>,
+	maxTries: number,
+): Promise<Response> {
+	let response = await fetcher();
+	let numTries = 1;
+
+	while (!response.ok && numTries < maxTries) {
+		response = await fetcher();
+		numTries++;
+	}
+	return response;
+}
```

Comment:
@tomrf1 I'll add a quick test for this function if we're happy with the general approach here, but wanted to see what you thought first. I've gone super simple. There's a bit of duplication mostly due to keeping typescript happy

Reply:
oh javascript actually has a `do...while` loop - this would be perfect!

Reply:
ahhh no I don't think it is :(

Reply:
Looks good. And if you enjoy recursion:
```
export const fetchWithRetry = (
    fetcher: () => Promise<Response>,
    maxRetries: number,
): Promise<Response> => fetcher().then(response => {
    if (!response.ok && maxRetries > 0) {
        return fetchWithRetry(fetcher, maxRetries-1);
    } else {
        return response;
    }
});
```

Reply:
ooo I think the recursive version is too good to pass on